### PR TITLE
WRP-862: Refactor Resizable to be hook-based implementation

### PR DESCRIPTION
**Pull request by juwonjeong**
Enact-DCO-1.0-Signed-off-by: Juwon Jeong (juwon.jeong@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Refactor `Resizable` to be hook-based implementation.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add `useResize` custom hook which returns a handler that indicates a resize is necessary.
Update `Resizable` to use `useResize` internally.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-862

### CODE REVIEW

File: packages/ui/Resizable/Resizable.js [hunk 1]
```diff
@@ -8,16 +8,9 @@
...
 
-/**
- * Used internally for things to notify children that they need to resize because of a parent
- * update.
- *
- * @type Object
- * @private
- */
-const ResizeContext = createContext();
+import ResizeContext from './ResizeContext';
```
Comment (MikyungKim) [inline packages/ui/Resizable/Resizable.js:13]: I expected `Resiable` to use `useResize`. Would you please consider it?

File: packages/ui/Resizable/useResizable.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import {useCallback, useContext, useEffect, useRef} from 'react';
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResizable.js:1]: Let's rename this to `useResize`.

File: packages/ui/Resizable/useResizable.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import ResizeContext from './ResizeContext';
+
+/**
+ * A custom hook which returns handler that indicates a resize is necessary
+ * @function useResizable
+ * @memberof ui/Resizable
+ * @ui
+ */
+const useResizable = (props, config) => {
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResizable.js:11]: Let's rename this component to `useResize`.

File: packages/ui/Resizable/useResizable.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	// `resize` is the name of the event on the component to listen for size changes.
+	const {resize} = config;
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResizable.js:13]: Where is `filter` from `Resizable` config?

File: packages/ui/Resizable/useResizable.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+			forwardHandler();
+		}
+		// Notifies a container that a resize is necessary
+		if (mutableRef.current.resizeRegistry) {
+			mutableRef.current.resizeRegistry.notify({action: 'invalidateBounds'});
+		}
+	}, [forwardHandler]);
+
+	const handlers = Object.assign({});
+	handlers[resize] = handleResize;
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResizable.js:46]:
```suggestion
	handlers[resize] = handleResize;

```

File: packages/ui/Resizable/useResizable.js (hunk 3 above)
Comment (juwonjeong) [reply packages/ui/Resizable/useResizable.js]: I renamed to useResize

File: packages/ui/Resizable/useResizable.js (hunk 4 above)
Comment (juwonjeong) [reply packages/ui/Resizable/useResizable.js]: I think I missed it. I added the logic of the filter.

File: packages/ui/Resizable/useResizable.js (hunk 5 above)
Comment (juwonjeong) [reply packages/ui/Resizable/useResizable.js]: It's added.

File: packages/ui/Resizable/useResize.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import {forward} from '@enact/core/handle';
+
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResize.js:2]:
```suggestion
```

File: packages/ui/Resizable/useResize.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import {useCallback, useContext, useEffect, useRef} from 'react';
+
+import ResizeContext from './ResizeContext';
+
+/**
+ * A custom hook which returns handler that indicates a resize is necessary
+ * @function useResize
+ * @memberof ui/Resizable
+ * @ui
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResize.js:11]: Let's rewrite this like 

```suggestion
 *
 * @param ...
 * @returns ...
 * @private
```

File: packages/ui/Resizable/useResize.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	// `resize` is the name of the event on the component to listen for size changes.
+	const {filter = null, resize = null} = config;
+
+	const resizeContextValue = useContext(ResizeContext);
+	const mutableRef = useRef({
+		resizeRegistry: null
+	});
+
+	useEffect(() => {
+		let {resizeRegistry} = mutableRef.current;
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResize.js:23]:
```suggestion
		const {resizeRegistry} = mutableRef.current;
```

**Comment by MikyungKim** [comment]
Would you please update the PR subject and description with the latest code?

File: packages/ui/Resizable/useResize.js (hunk 8 above)
Comment (juwonjeong) [reply packages/ui/Resizable/useResize.js]: It's fixed.

File: packages/ui/Resizable/useResize.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+import ResizeContext from './ResizeContext';
+
+/**
+ * Object returned by `useResize`
+ * Object has a property named with the value of the resize property of config.
+ * And this property has resizehandler.
+ *
+ * @typedef {Object} useResizeInterface
+ * @memberof ui/Resizable
```
Comment (MikyungKim) [inline packages/ui/Resizable/useResize.js:12]: Does below make sense?
```suggestion
 * @property {Function} resize The handler for resize
 * @memberof ui/Resizable
```

**Comment by MikyungKim** [comment]
Let's see the travis issue on node 20 separately, if node 19 runs ok, let's fix the version to 19 instead of node temporarily.
cc. @0x64

File: .travis.yml [hunk 10]
```diff
@@ -11,10 +11,14 @@ install:
     - npm install -g codecov
     - git clone --branch=develop --depth 1 https://github.com/enactjs/cli ../cli
     - pushd ../cli
+    - npm cache verify
```
Comment (MikyungKim) [inline .travis.yml:14]: Let's fix the version to 19 for now.

**Review by MikyungKim — APPROVED** [review]
LGTM

File: packages/ui/Resizable/useResize.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+		// Notifies a container that a resize is necessary
+		if ((filter === null || filter(ev, props)) && mutableRef.current.resizeRegistry) {
+			mutableRef.current.resizeRegistry.notify({action: 'invalidateBounds'});
+		}
+	}, [filter, props, resize]);
+
+	const handlers = Object.assign({});
+	handlers[resize] = handleResize;
+
+	return handlers;
```
Comment (0x64) [inline packages/ui/Resizable/useResize.js:60]:
```suggestion
	return {[resize]: handleResize};
```
What about this?

Comment (juwonjeong) [reply packages/ui/Resizable/useResize.js]: It's fixed.

**Review by 0x64 — APPROVED** [review]

**Pull request merged**